### PR TITLE
Elements that already exists in the set should not be added again

### PR DESCRIPTION
**Pull request by spydon**
To fulfill one of the premises of being a set.

### CODE REVIEW

File: lib/ordered_set.dart [hunk 1]
```diff
@@ -72,24 +75,23 @@ class OrderedSet<E> extends IterableMixin<E> implements Iterable<E> {
   ///
   /// Since elements are always added, this should always return the length of
   /// [elements].
```
Comment (luanpotter) [inline lib/ordered_set.dart]: fix doc

File: test/ordered_set_test.dart [hunk 2]
```diff
@@ -111,23 +111,23 @@ void main() {
 
       test('with identical elements', () {
         final a = OrderedSet<int>();
-        expect(a.addAll([4, 3, 3, 2, 2, 2, 1]), 7);
-        expect(a.length, 7);
-        expect(a.toList().join(), '1222334');
+        expect(a.addAll([4, 3, 3, 2, 2, 2, 1]), 4);
+        expect(a.length, 4);
+        expect(a.toList().join(), '1234');
```
Comment (luanpotter) [inline test/ordered_set_test.dart:116]: I think some of these tests were intended to test the main premise of ordered_set: that multiple elements with the same priority can be added. maybe we should have a new test for that with non-identical but same-priority elements?

Comment (spydon) [reply test/ordered_set_test.dart:116]: There are tests for that too, check line 100 for example.

Comment (spydon) [reply test/ordered_set_test.dart:116]: And this test actually tests that too since they are all priority 0 right?

Comment (spydon) [reply test/ordered_set_test.dart:116]: Added another test for that.

**Review by luanpotter — APPROVED** [review]
:shipit:

**Pull request merged**